bake/installation-configuration.sh

```diff
@@ -97,6 +97,54 @@ then
   oc delete node "$(oc get nodes -ojsonpath='{.items[?(@.metadata.name != "'"$(hostname)"'"].metadata.name}')"
 fi
 
+verify_csr_subject() {
+  local csr=${1}
+  local subject
+
+  subject="$(oc get csr -ojsonpath='{.spec.request}' "${csr}" |base64 -d | openssl req -noout -subject -nameopt multiline)"
+
+  if [ "$(echo "${subject}" |grep commonName |awk '{print $3}')" != "system:node:$(hostname)" ]
+  then
+    echo "CommonName is not 'system:node:master1'"
+    return
+  fi
+
+  if [ "$(echo "${subject}" |grep organizationName |awk '{print $3}')" != "system:nodes" ]
+  then
+    echo "Organization is not 'system:nodes'"
+    return
+  fi
+}
+
+# If the kubelet API server client certificate has expired:
+#   1. wait for the respective CSR to be created
+#   2. verify it
+#   3. approve it
+#   4. wait for the certificate to be issued
+KUBELET_CLIENT_CERTIFICATE=/var/lib/kubelet/pki/kubelet-client-current.pem
+until openssl x509 -in ${KUBELET_CLIENT_CERTIFICATE} -checkend 30 &> /dev/null
+do
+  echo "${KUBELET_CLIENT_CERTIFICATE} has expired, waiting for new one to be issued..."
+
+  csr=$(oc get csr -o go-template='{{range .items}}{{if and (not .status) (eq .spec.signerName "kubernetes.io/kube-apiserver-client-kubelet") (eq .spec.username "system:serviceaccount:openshift-machine-config-operator:node-bootstrapper")}}{{.metadata.name}}{{"\n"}}{{end}}{{end}}' | head -1 || true)
+  if [ -z "${csr}" ]
+  then
+    sleep 5
+    continue
+  fi
+
+  echo "${csr} is pending. Verifying CSR before approving it..."
+  err=$(verify_csr_subject "${csr}")
+  if [ -n "${err}" ]
+  then
+    echo "${csr} could not be verified: ${err}"
+    break
+  fi
+
+  echo "${csr} successfully verified. Approving it..."
+  oc adm certificate approve "${csr}"
+done
+
```

Comment:
Consider adding a log here saying: `The kubelet API server client certificate is valid`

Reply:
This makes the logs quite clearer, thanks! I added such a log for both the `kubelet` client and serving certs.